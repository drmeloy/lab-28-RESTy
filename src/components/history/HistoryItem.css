.HistoryItem {
  border: 1px solid;
  padding: 0 15px;
  margin: 15px 15px;
  font-family: 'Oswald', sans-serif;
  background-image: url('../../assets/trianglify-lowres\ \(3\).png');
  background-size: cover;
  word-wrap: break-word;
  cursor: pointer;
}