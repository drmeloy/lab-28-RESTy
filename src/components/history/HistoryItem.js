import React from 'react';
import PropTypes from 'prop-types';
import styles from './HistoryItem.css';

const HistoryItem = ({ method, url, requestBody }) => {  
return (
    <div className={styles.HistoryItem}>
      <p className='method'>METHOD: {method}</p>
      <p className='url'>URL: {url}</p>
      <p className='request-body'>BODY: {requestBody}</p>
    </div>
)};

HistoryItem.propTypes = {
  method: PropTypes.string.isRequired,
  url: PropTypes.string.isRequired,
  requestBody: PropTypes.string
};

export default HistoryItem;
