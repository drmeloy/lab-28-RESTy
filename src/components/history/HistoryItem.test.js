import React from 'react';
import { shallow } from 'enzyme';
import HistoryItem from './HistoryItem';

describe('HistoryItem component', () => {
  it('renders HistoryItem', () => {
    const wrapper = shallow(<HistoryItem />);
    expect(wrapper).toMatchSnapshot();
  });
});
