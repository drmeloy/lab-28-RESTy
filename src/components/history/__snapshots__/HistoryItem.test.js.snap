// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HistoryItem component renders HistoryItem 1`] = `
<div
  className="HistoryItem"
  onClick={[Function]}
>
  <p
    className="method"
  >
    METHOD: 
  </p>
  <p
    className="url"
  >
    URL: 
  </p>
  <p
    className="request-body"
  >
    BODY: 
  </p>
</div>
`;
